### PR TITLE
Fixing IsTypeOf() result error.

### PR DESCRIPTION
**Pull request by Y-way**

### CODE REVIEW

File: Source/Urho3D/Core/Object.cpp [hunk 1]
```diff
@@ -58,10 +58,15 @@ bool TypeInfo::IsTypeOf(StringHash type) const
...
+    if (typeInfo == nullptr)
+    {
+        return false;
+    }
+    
     const TypeInfo* current = this;
     while (current)
     {
-        if (current == typeInfo)
+        if (current->GetType() == typeInfo->GetType())
```
Comment (weitjong) [inline Source/Urho3D/Core/Object.cpp]: Perhaps `current == typeInfo || current->GetType() == typeInfo->GetType()`

File: Source/Urho3D/Core/Object.cpp [hunk 2]
```diff
@@ -58,10 +58,15 @@ bool TypeInfo::IsTypeOf(StringHash type) const
 
 bool TypeInfo::IsTypeOf(const TypeInfo* typeInfo) const
 {
+    if (typeInfo == nullptr)
+    {
+        return false;
+    }
```
Comment (weitjong) [inline Source/Urho3D/Core/Object.cpp]: Usually for one liner we don't put it inside a code block.

**Comment by Y-way** [comment]
IMO, `IsTypeOf(StringHash type)` is not safe yet, because the different strings maybe have the same StringHash, although the probability is extremely low.

**Comment by weitjong** [comment]
If you cannot trust StringHash in this comparison then you might as well just don't use it and just compare the string directly.

**Comment by eugeneko** [comment]
> because the different strings maybe have the same StringHash

@Y-way if you accidentally get such classes in your code, Urho as whole will fall apart.
You can assume that StringHash-es for type names never collide.
Fallback to string comparison make this function almost useless because you are basically re-implementing standard `dynamic_cast`.

**Comment by Y-way** [comment]
@weitjong @eugeneko Yes, you are right. So we did not discuss how to improve it.Right.
There is no problem at present.

**Comment by weitjong** [comment]
Actually I believe both Eugene and me are basically saying the same thing. To me, the String comparison following the StringHash comparison looks a little superfluous and I don't recall I ever see it being done elsewhere in the code. If the current StringHash could not be trusted then we need to fix the StringHash, it can be trusted as far as I know.

**Pull request merged**